pennylane/transforms/decompositions/single_qubit_unitary.py

```diff
@@ -81,27 +71,15 @@ def zyz_decomposition(U, wire):
     """
     U = _convert_to_su2(U)
 
-    # Check if the matrix is diagonal; only need to check one corner.
-    # If it is diagonal, we don't need a full Rot, just return an RZ.
-    if math.allclose(U[0, 1], [0.0]):
-        omega = 2 * math.angle(U[1, 1])
-        return [qml.RZ(omega, wires=wire)]
-
-    # If the top left element is 0, can only use the off-diagonal elements. We
-    # have to be very careful with the math here to ensure things that get
-    # multiplied together are of the correct type in the different interfaces.
-    if math.allclose(U[0, 0], [0.0]):
-        phi = 0.0
-        theta = -np.pi
-        omega = 1j * math.log(U[0, 1] / U[1, 0]) - np.pi
-    else:
-        # If not diagonal, compute the angle of the RY
-        cos2_theta_over_2 = math.abs(U[0, 0] * U[1, 1])
-        theta = 2 * math.arccos(math.sqrt(cos2_theta_over_2))
-
-        el_division = U[0, 0] / U[1, 0]
-        tan_part = math.cast_like(math.tan(theta / 2), el_division)
-        omega = 1j * math.log(tan_part * el_division)
-        phi = -omega - math.cast_like(2 * math.angle(U[0, 0]), omega)
-
-    return [qml.Rot(math.real(phi), math.real(theta), math.real(omega), wires=wire)]
+    # Derive theta from the diagonal elements; arcsin more accurate than arccos
```

Comment:
@josh146 I managed to hugely simplify the `zyz_decomposition` and eliminate all the checks. The problem, now, is that the precision is reduced and a number of the two-qubit decomposition checks are failing because their matrices are specified only to 8 decimal places (the new decomposition gives equivalence for them only up to ~1e-4). I guess these were special cases previously, and the checks helped produce more accurate results (like an `RZ` instead of a `Rot`).

Reply:
I tried a couple different configurations as well, and something that was interesting is that `arcsin` seemed to produce better results than `arccos`, but also failed more spectacularly because the precision would produce values greater than 1 (which `arccos` can't handle either, but for some reason it did not have the same issue)